test/fakeintake/client/client.go

```diff
@@ -725,6 +725,26 @@ func (c *Client) GetLastProcessPayloadAPIKey() (string, error) {
 	return payloads[len(payloads)-1].APIKey, nil
 }
 
+// GetAllProcessPayloadAPIKeys fetches fakeintake on `/api/v1/collector` endpoint and returns
+// a list of unique API keys of the received process payloads
+func (c *Client) GetAllProcessPayloadAPIKeys() ([]string, error) {
+	payloads, err := c.getFakePayloads(processesEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	keysFound := make(map[string]struct{})
+	keys := make([]string, 0)
```

Comment:
Curious what you think about pre-allocating capacity of `keys` to `length(payloads)`. I'm not too sure what the distribution of duplicate apiKeys looks like (i.e if there are many duplicates this may be overkill since you end up using less space than capacity, but if there are rarely duplicates this may save on allocations during append).

Reply:
I thought about it, but it would be overkill. I would say the normal case is to have one api key for all payloads, but the test I use it in, I would expect two.  Since the assert clauses run for about 20 minutes, there would be 20 payloads in total. Also, this function is used pretty sparingly, so I don't think it would make a difference regardless.